src/WebJobs.Script.WebHost/Security/DefaultKeyValueConverterFactory.cs

```diff
@@ -7,15 +7,51 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using static Microsoft.Azure.Web.DataProtection.Constants;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public sealed class DefaultKeyValueConverterFactory : IKeyValueConverterFactory
     {
-        private static readonly PlaintextKeyValueConverter PlainTextValueConverter = new PlaintextKeyValueConverter(FileAccess.ReadWrite);
+        private bool _encryptionSupported;
+        private static readonly PlaintextKeyValueConverter PlaintextValueConverter = new PlaintextKeyValueConverter(FileAccess.ReadWrite);
 
-        public IKeyValueReader GetValueReader(Key key) => PlainTextValueConverter;
+        public DefaultKeyValueConverterFactory()
+        {
+            _encryptionSupported = IsEncryptionSupported();
+        }
 
-        public IKeyValueWriter GetValueWriter(Key key) => PlainTextValueConverter;
+        private static bool IsEncryptionSupported()
+        {
+            if (WebScriptHostManager.IsAzureEnvironment)
+            {
+                // We're temporarily placing encryption behind a feature toggle until
+                // other consumers (e.g. portal) are updated to work with it.
+                // TODO: Remove this
+                return string.Equals(Environment.GetEnvironmentVariable("AzureWebJobsEncryptionEnabled"), "true", StringComparison.OrdinalIgnoreCase);
+            }
+
+            return Environment.GetEnvironmentVariable(AzureWebsiteLocalEncryptionKey) != null;
```

Comment:
So for people outside of Functions (e.g. continuous WebJobs / ScriptHost scenarios), they'd use a local key? How would they do key management - e.g. add a new key and have it encrypted? In such scenarios there is no portal and keys are managed manually.